x/dex/cache/cache.go

```diff
@@ -17,18 +17,16 @@ const SynchronizationTimeoutInSeconds = 5
 type MemState struct {
 	storeKey sdk.StoreKey
 
-	contractsToProcess      *datastructures.SyncSet[string]
-	contractsToDepsMtx      *sync.Mutex
-	contractsToDependencies *datastructures.TypedSyncMap[string, []string]
+	// TODO: all of these need to be in the memstore
+	contractsToProcessMtx *sync.RWMutex
```

Comment:
do we still need these mutex? i think we have mutex in front of kvstore.Set/Get etc., or maybe that's just the iavl store not the memstore?

Reply:
Yeah i think we _should_ be able to remove these mutexes, I wanted to leave them for safety since i wasnt confident in the expected behavior of the concurrent processing for dex end block. I can add a todo to consider removing them, but don't want to remove them at the moment so that any issues can be more clearly attributed to OCC issues